airflow/models/taskinstance.py

```diff
@@ -1808,13 +1808,13 @@ def handle_failure(
             except Exception:
                 self.log.exception("Failed to send email to: %s", task.email)
 
-        if callback and context:
-            self._run_finished_callback(callback, context, callback_type)
-
         if not test_mode:
             session.merge(self)
             session.flush()
 
+        if callback and context:
+            self._run_finished_callback(callback, context, callback_type)
+
```

Comment:
Can you add test? Looks like something that can be tested

Reply:
I can try; that's all I can promise 🙂

Reply:
@sydykov, any update on this?

Reply:
@ephraimbuddy, hi! I am planning to get back to this in January. Thank you for asking, and I regret delaying the PR this much.

Reply:
@ephraimbuddy, hello. I am still traveling and haven't had a chance to get back to this yet, unfortunately.